### ISSUE TITLE
[Distributed Index] Empty doc returned to continuous _changes feed - bulk set issue

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1465]
"Couchbase Sync Gateway/feature/issue_1440(a4b0eb1)"

```
6-01-14T15:11:56.972-08:00 Changes+: DocChanged - 419e4203-7d1e-48cd-bbeb-63ca22869c5d
2016-01-14T15:11:56.972-08:00 Changes+: DocChanged - 0c1fce22-83a0-4928-a791-3ac034149b5e
2016-01-14T15:11:56.973-08:00 Changes+: DocChanged - 87f6282c-2e22-4cef-aab0-142fe6dd722d
2016-01-14T15:11:56.973-08:00 Changes+: DocChanged - a9d3a516-a18d-430b-8524-700abdae3f20
2016-01-14T15:11:56.973-08:00 Changes+: DocChanged - 1ac57822-843b-42e3-aaca-c019c981b3c8
2016-01-14T15:11:56.973-08:00 Changes+: DocChanged - 55d5ff10-8205-4a92-9547-b9b98c8bb3d4
2016-01-14T15:11:56.991-08:00 WARNING: Non-recoverable error doing insert during bulk set: Key already exists. -- base.CouchbaseBucketGoCB.processBulkSetEntriesBatch() at bucket_gocb.go:280
2016-01-14T15:11:56.992-08:00 WARNING: Non-recoverable error doing insert during bulk set: Key already exists. -- base.CouchbaseBucketGoCB.processBulkSetEntriesBatch() at bucket_gocb.go:280
2016-01-14T15:11:57.021-08:00 Changes+: DocChanged - 8ca888d3-0a2f-45ac-9c79-7a22139eaa84
2016-01-14T15:11:57.021-08:00 Changes+: DocChanged - abdab767-3be6-4e22-9446-0d3c06232ba3
2016-01-14T15:11:57.021-08:00 Changes+: DocChanged - 8aa051ba-5cdf-4fb5-a99e-7422a2e83e62
2016-01-14T15:11:57.021-08:00 Changes+: DocChanged - b89fc515-80c8-4d60-8918-5885a93d80d6
2016-01-14T15:11:57.029-08:00 Changes+: DocChanged - ce12a458-bb6a-495b-b2b3-8b9ba88318d6
2016-01-14T15:11:57.034-08:00 Changes+: DocChanged - 086a29b0-984e-4f7b-a46d-077e9fe743db
2016-01-14T15:11:57.036-08:00 Changes+: DocChanged - e511ec4a-6056-466f-98f4-c62a66177f67
2016-01-14T15:11:57.036-08:00 Changes+: DocChanged - 1059e0bf-1993-4142-9340-bdc765db63a4
2016-01-14T15:11:57.045-08:00 Changes+: DocChanged - b02fc180-ae7c-4295-9b6d-f26098592c78
```

**Comment by sethrosetter** [comment]
Full logs - https://s3.amazonaws.com/sync-gateway-logs/functional_tests-test_continuous.py-test_continuous_changes_parametrized%5BDI-1%5D-2016-01-14-15-12-19-sglogs.zip

**Comment by adamcfraser** [comment]
Based on the error, in the log, this looks like the following:
1. One or two entries in a bulk set operation are timing out waiting for a response from Couchbase Server
2. Timeout is a recoverable error, so we retry the set for those entries.
3. The original requests that timed out eventually succeed on the server
4. The retry fails with a 'Key already exists'

**Comment by adamcfraser** [comment]
Looking closer at the failing cases, the SG logs show both writer nodes attempting to update the same channel index block, for both channel ABC and channel *.  

Needs further investigation to understand whether the problem is that the index partition map isn't aligned with CBGT's partitions (so each node is trying to write distinct vbuckets to the same partition block), or if both feeds are getting entries for the same vbucket.

This shouldn't cause an error in any case - it should just be performance degradation at worst.

**Comment by adamcfraser** [comment]
Fixed with #1440.
